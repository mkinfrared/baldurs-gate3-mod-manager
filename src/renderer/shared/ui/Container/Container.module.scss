.Container {
  margin-inline: auto;
  max-width: 106rem;
}
