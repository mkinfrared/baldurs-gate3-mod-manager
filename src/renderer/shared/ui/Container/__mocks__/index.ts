export { default } from "./Container.mock";
