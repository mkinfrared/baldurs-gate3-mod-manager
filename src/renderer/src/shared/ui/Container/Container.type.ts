import { ReactNode } from "react";

export type ContainerProps = {
  children?: ReactNode;
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
