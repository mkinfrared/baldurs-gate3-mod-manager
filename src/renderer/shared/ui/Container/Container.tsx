import { classNames } from "renderer/shared/lib/helpers";

import css from "./Container.module.scss";
import { ContainerProps } from "./Container.type";

const Container = ({ children, className }: ContainerProps) => (
  <div className={classNames(css.Container, className)} data-testid="Container">
    {children}
  </div>
);

export default Container;
