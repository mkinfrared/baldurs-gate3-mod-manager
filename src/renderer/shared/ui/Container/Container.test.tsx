import { render } from "@testing-library/react";

import Container from "./Container";

describe("<Container />", () => {
  const Component = <Container />;

  it("should be defined", () => {
    expect(Container).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Container");

    expect(element).toBeDefined();
  });
});
