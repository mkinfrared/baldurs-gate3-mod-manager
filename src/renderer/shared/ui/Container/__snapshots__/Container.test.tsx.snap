// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match the snapshot 1`] = `
<div>
  <div
    class="Container"
    data-testid="Container"
  />
</div>
`;
